providers/dns/azure/azure.go

```diff
@@ -91,59 +121,55 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 	rec := dns.RecordSet{
 		Name: &relative,
 		RecordSetProperties: &dns.RecordSetProperties{
-			TTL:        to.Int64Ptr(60),
+			TTL:        to.Int64Ptr(int64(d.config.TTL)),
 			TxtRecords: &[]dns.TxtRecord{{Value: &[]string{value}}},
 		},
 	}
 
-	_, err = rsc.CreateOrUpdate(d.context, d.resourceGroup, zone, relative, dns.TXT, rec, "", "")
-	return err
-}
-
-// Returns the relative record to the domain
-func toRelativeRecord(domain, zone string) string {
-	return acme.UnFqdn(strings.TrimSuffix(domain, zone))
+	_, err = rsc.CreateOrUpdate(ctx, d.config.ResourceGroup, zone, relative, dns.TXT, rec, "", "")
+	return fmt.Errorf("azure: %v", err)
 }
 
 // CleanUp removes the TXT record matching the specified parameters
 func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
+	ctx := context.Background()
```

Comment:
It should accept a context rather than using background.

Reply:
it's not possible because this method is a part of an interface